src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  };
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
-}
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  };
+  hide() {
+    this.hidden = !this.hidden;
+  };
+};
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(victim) {
+    if (victim instanceof Carnivore || victim.hidden) {
+      return;
+    } else {
+      victim.health -= 50;
+    }
+
+    if (victim.health === 0) {
+      const index = Animal.alive.indexOf(victim);
```

Comment:
What if `indexOf` returns -1, what `splice` will delete ?

Let's add a specific check.

Reply:
added check for index >= 0